src/V3Gate.cpp

```diff
@@ -167,86 +125,476 @@ class GateLogicVertex final : public GateEitherVertex {
+    }
+
+    void visit(AstAlwaysPublic* nodep) override { iterateLogic(nodep, true, "AlwaysPublic"); }
+    void visit(AstCFunc* nodep) override {  //
+        iterateLogic(nodep, nodep->slow(), "C Function", "C Function");
+    }
+    void visit(AstNodeProcedure* nodep) override {
+        const bool slow = VN_IS(nodep, Initial) || VN_IS(nodep, Final);
+        iterateLogic(nodep, slow, nodep->isJustOneBodyStmt() ? nullptr : "Multiple Stmts");
+    }
+    void visit(AstAssignAlias* nodep) override {  //
+        iterateLogic(nodep);
+    }
+    void visit(AstAssignW* nodep) override {  //
+        iterateLogic(nodep);
+    }
+    void visit(AstCoverToggle* nodep) override {
+        iterateLogic(nodep, false, "CoverToggle", "CoverToggle");
+    }
+    void visit(AstSenItem* nodep) override {
+        VL_RESTORER(m_inSenItem);
+        m_inSenItem = true;
+        if (m_logicVertexp) {  // Already under logic, e.g.: AstEventControl
+            iterateChildrenConst(nodep);
+        } else {  // Standalone item, under a SenTree or an Active
+            iterateLogic(nodep, false, nullptr, "senItem");
+        }
+    }
+    void visit(AstNodeVarRef* nodep) override {
+        if (!m_logicVertexp) return;
+
+        AstVarScope* const vscp = nodep->varScopep();
+        GateVarVertex* const vVtxp = m_graphp->makeVarVertex(vscp);
+
+        if (m_inSenItem) {
+            vVtxp->setIsClock();
+            vscp->user2(true);
+        } else if (m_inClockedActive && nodep->access().isReadOnly()) {
+            // For SYNCASYNCNET
+            if (vscp->user2()) {
+                if (!vVtxp->rstAsyncNodep()) vVtxp->rstAsyncNodep(nodep);
+            } else {
+                if (!vVtxp->rstSyncNodep()) vVtxp->rstSyncNodep(nodep);
+            }
+        }
+
+        // We use weight of one; if we ref the var more than once, when we simplify,
+        // the weight will increase
+        if (nodep->access().isWriteOrRW()) m_graphp->addEdge(m_logicVertexp, vVtxp, 1);
+        if (nodep->access().isReadOrRW()) m_graphp->addEdge(vVtxp, m_logicVertexp, 1);
+    }
+    void visit(AstConcat* nodep) override {
+        UASSERT_OBJ(!(VN_IS(nodep->backp(), NodeAssign)
+                      && VN_AS(nodep->backp(), NodeAssign)->lhsp() == nodep),
+                    nodep, "Concat on LHS of assignment; V3Const should have deleted it");
+        iterateChildrenConst(nodep);
+    }
+
+    //--------------------
+    void visit(AstNode* nodep) override {
+        if (m_logicVertexp) checkNode(nodep);
+        iterateChildrenConst(nodep);
+    }
+
+    // CONSTRUCTORS
+    explicit GateBuildVisitor(AstNetlist* nodep) { iterateChildrenConst(nodep); }
+
+public:
+    static std::unique_ptr<GateGraph> apply(AstNetlist* netlistp) {
+        return std::unique_ptr<GateGraph>{GateBuildVisitor{netlistp}.m_graphp};
+    }
+};
+
+//######################################################################
+// SYCNASYNC warning
+
+static void warnSyncAsync(GateGraph& graph) {
```

Comment:
Ok, I will, but I'm curious why. `static` file (or namespace) scope functions are local to the compilation unit and won't conflict at link time. IMO the unnecessary prefixes make things harder to read. Not an issue for a single function like this, but if you have many smaller ones it can cause clutter. Are you worried about gdb breakpoint ambiguity?

Reply:
Yes, exactly, also profile reports and such I look only at the function name and want it to be meaningful without looking at the filename. Thanks.